docs/read_and_write.rst

```diff
@@ -339,3 +339,25 @@ The ability to achieve fast random access to individual rows plays a crucial rol
 such as random sampling and shuffling in ML training.
 Additionally, it empowers users to construct secondary indices,
 enabling swift execution of queries for enhanced performance.
+
+
+Concurrent Writer on S3
```

Comment:
Could you merge this section with: https://github.com/lancedb/lance/blob/e789c74f4518d95c48bad0a53dbfb42c21923f52/docs/read_and_write.rst#committing-mechanisms-for-s3